auth/src/main/java/com/firebase/ui/auth/ui/email/CheckEmailFragment.java

```diff
@@ -94,6 +96,52 @@ public static CheckEmailFragment getInstance(@NonNull FlowParameters flowParamet
         return fragment;
     }
 
+    public static void checkAccountExists(Activity activity,
+                                          final BaseHelper helper,
+                                          final CheckEmailListener listener,
+                                          @NonNull final String email,
+                                          @Nullable final String name,
+                                          @Nullable final Uri photoUri) {
```

Comment:
There are a few things I don't really like here:

  * Taking an instance of  `BaseHelper` so that we can start/stop the progress dialog on behalf of the caller does not seem right.  This method should not be concerned with UI.
   * We take in `name` and `photoUri` only to give it right back
   * `CheckEmailListener` really doesn't seem like it saves much code.  For example:

```java
if (providers == null || providers.isEmpty()) {
    // New user
} else if (EmailAuthProvider.PROVIDER_ID.equalsIgnoreCase(providers.get(0))) {
   // Email user
} else {
   // IDP user
}
```
becomes

```java
@Override
onNewUser(User user) {

}

@Override
onExistingEmailUser(User user) {

}

@Override
onExistingIdpUser(User user) {

}
```

So what if we defined this method to be like this (coding in a comment field so excuse typos):

```java
public static Task<String> getTopProvider(FirebaseAuth auth, String email) {
   TaskCompletionSource<String> source = new TaskCompletionSource<>();

   auth.fetchProvidersForEmail(email)
       .addOnCompleteListener(Task<ProviderQueryResult> task) {
          if (!task.isSuccessful) {
              source.setException(task.getException());
              return;
          }

           List<String> providers = task.getResult().getProviders();
           if (providers == null || providers.isEmpty()) {
               return null;
           } else {
               return providers.get(0);
           }
       });

    return source.getTask();
}
```

and then consumers can just call this `Task` and branch on if the result is `EmailAuthProvider.PROVIDER_ID`, `null` or some IDP.